TicketBookingSystemNew.sol

```diff
@@ -0,0 +1,357 @@
+    mapping(uint256 => TicketInfo) public ticketInfoMapping;
+
+    struct TicketInfo {
+        string showTitle;
+        uint256 showDate;
+        uint256 seatRow;
+        uint256 seatNumber;
+        address mintedBy;
+    }
+
+    constructor() ERC721("Ticket", "TKT") {
+        minterAddress = msg.sender;
+        tokenId = 0;
+    }
+
+    //this function mints a new ticket, and saves information about ticket info in ticketInfoMapping
+    //against the tokenId
+    function mintTicket(
+        address receiver,
+        string memory _showTitle,
+        uint256 _showDate,
+        uint256 _seatRow,
+        uint256 _seatNumber,
+        address _mintedBy
+    ) external onlyOwner returns (uint256) {
+        ticketInfoMapping[tokenId] = TicketInfo({
+            showTitle: _showTitle,
+            showDate: _showDate,
+            seatRow: _seatRow,
+            seatNumber: _seatNumber,
+            mintedBy: _mintedBy
+        });
+
+        uint256 newTokenId = tokenId;
+        _safeMint(receiver, newTokenId);
+        tokenId += 1;
+        return newTokenId;
+    }
+
+    //helper function to get the address who minted the ticket
+    function mintedBy(uint256 _tokenId) external view returns (address) {
+        return ticketInfoMapping[_tokenId].mintedBy;
+    }
+
+    //helper function to get show data from ticket id
+    function getShowDateFromTicketId(uint256 _tokenId)
+        external
+        view
+        returns (uint256)
+    {
+        return ticketInfoMapping[_tokenId].showDate;
+    }
+
+    //check if ticket exists
+    function exists(uint256 _tokenId) public view returns (bool) {
+        return _exists(_tokenId);
+    }
+
+    //burn the ticket
+    function burn(uint256 tokenId) public {
+        _burn(tokenId);
+    }
+}
+
+contract Poster is ERC721, Ownable {
+    uint256 private posterTokenId;
+
+    constructor() ERC721("Poster", "PSTR") {
+        uint256 posterTokenId = 0;
+    }
+
+    //mint new poster
+    function mintNewPoster(address _to) external onlyOwner returns (uint256) {
+        uint256 newTokenId = posterTokenId;
+        _safeMint(_to, newTokenId);
+        posterTokenId += 1;
+        return posterTokenId;
+    }
+}
+
+contract TicketBookingSystem {
+    address payable public salesManager;
+    address public ticketBuyer;
+    string constant _seatViewLink = "https://seatplan.com/";
+    uint256 numberOfSeatsPerRow = 10;
+    uint256 numberOfRows = 2;
+    uint256 totalNumberOfSeats = numberOfSeatsPerRow * numberOfRows;
+    uint256 availableTickets = totalNumberOfSeats;
+    uint256 showDate;
+    string showTitle;
+    uint256 price;
+    ShowStatus showStatus;
+    Ticket ticketInstance;
+    Poster posterInstance;
+
+    //this mapping stores amount to be refunded against each address if show gets cancelled
+    mapping(address => uint256) amountToBeRefunded;
+
+    //this mapping stores a Seat struct against each seat index (1-60)
+    mapping(uint256 => Seat) public seats;
```

Comment:
As the number of rows are 2 and the number of seats per row is 10, should the seat index be a number between 1 and 20? Nothing wrong, just asking for a change in the comment.

Reply:
yes, since 3 rows and 20 seats per row were taking longer to deploy the SC, i changed the values while debugging.
we can update the comment to 1-20